doc/dhcp_server/port_based_dhcp_server_high_level_design.md

```diff
@@ -0,0 +1,1045 @@
+# IPv4 Port Based DHCP_SERVER in SONiC
+# High Level Design Document
+**Rev 0.1**
+
+# Table of Contents
+<!-- TOC -->
+
+- [IPv4 Port Based DHCP_SERVER in SONiC](#ipv4-port-based-dhcp_server-in-sonic)
+- [High Level Design Document](#high-level-design-document)
+- [Table of Contents](#table-of-contents)
+- [Revision](#revision)
+- [About this Manual](#about-this-manual)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+                    - [Table 1: Abbreviations](#table-1-abbreviations)
+                    - [Table 2: Definitions](#table-2-definitions)
+- [Overview](#overview)
+    - [Background](#background)
+    - [Functional Requirements](#functional-requirements)
+    - [Configuration and Management Requirements](#configuration-and-management-requirements)
+- [Design](#design)
+    - [Design Overview](#design-overview)
+    - [Container](#container)
+    - [DHCP Server Daemon](#dhcp-server-daemon)
+        - [Generate Config](#generate-config)
+        - [Update Lease](#update-lease)
+    - [Customize DHCP Packet Options](#customize-dhcp-packet-options)
+    - [DHCP Relay Daemon](#dhcp-relay-daemon)
+    - [DB Changes](#db-changes)
+        - [Config DB](#config-db)
+            - [DB Objects](#db-objects)
+            - [Yang Model](#yang-model)
+        - [State DB](#state-db)
+            - [DB Objects](#db-objects)
+            - [Yang Model](#yang-model)
+    - [Flow Diagrams](#flow-diagrams)
+        - [DHCP Server Flow](#dhcp-server-flow)
+        - [Config Change Flow](#config-change-flow)
+        - [Lease Update Flow](#lease-update-flow)
+        - [Start flow of dhcprelayd](#start-flow-of-dhcprelayd)
+        - [Work flow of dhcprelayd capture db change](#work-flow-of-dhcprelayd-capture-db-change)
+- [CLI](#cli)
+    - [Config CLI](#config-cli)
+    - [Show CLI](#show-cli)
+- [Test](#test)
+    - [Unit Test](#unit-test)
+        - [Config CLI](#config-cli)
+        - [Show CLI](#show-cli)
+        - [Others](#others)
+    - [Test Plan](#test-plan)
+
+<!-- /TOC -->
+
+
+# Revision
+
+| Rev |     Date    |       Author       | Change Description                  |
+|:---:|:-----------:|:-------------------|:-----------------------------------|
+| 0.1 |  2023/02/08 | Yaqiang Zhu, <br> Jing Kan       | Initial version                     |
+
+# About this Manual
+
+This document describes the design details of **ipv4 port based DHCP server** feature.
+Dynamic Host Configuration Protocol (DHCP) server is used to centrally manage and configure IP addresses of clients dynamically. When server receives requests from DHCP-enabled clients, it would offer information to client about IP address, subnet mask, gateway etc.
```

Comment:
If you could mention the exact use-case for this feature, would be helpful.

Reply:
Added, pls refer to pic in `Design considerations` section